src/integrationTest/java/org/opensearch/security/FlsDlsAndFieldMaskingTest.java

```diff
@@ -0,0 +1,946 @@
+			String firstSongId = FIRST_INDEX_ID_SONG_3;
+			Song firstSong = FIRST_INDEX_SONGS_BY_ID.get(firstSongId);
+			String secondSongId = SECOND_INDEX_ID_SONG_4;
+			Song secondSong = SECOND_INDEX_SONGS_BY_ID.get(secondSongId);
+
+			MultiSearchRequest request = new MultiSearchRequest();
+			request.add(queryByIdsRequest(FIRST_INDEX_NAME, firstSongId));
+			request.add(queryByIdsRequest(SECOND_INDEX_NAME, secondSongId));
+			MultiSearchResponse response = restHighLevelClient.msearch(request, DEFAULT);
+
+			assertThat(response, isSuccessfulMultiSearchResponse());
+			assertThat(response, numberOfSearchItemResponsesIsEqualTo(2));
+
+			MultiSearchResponse.Item[] responses = response.getResponses();
+
+			assertThat(responses[0].getResponse(), allOf(
+					searchHitsContainDocumentWithId(0, FIRST_INDEX_NAME, firstSongId),
+					searchHitContainsFieldWithValue(0, FIELD_TITLE, firstSong.getTitle()),
+					searchHitContainsFieldWithValue(0, FIELD_LYRICS, VALUE_TO_MASKED_VALUE.apply(firstSong.getLyrics())),
+					searchHitContainsFieldWithValue(0, FIELD_ARTIST, VALUE_TO_MASKED_VALUE.apply(firstSong.getArtist())),
+					searchHitContainsFieldWithValue(0, FIELD_STARS, firstSong.getStars())
+			));
+			assertThat(responses[1].getResponse(), allOf(
+					searchHitsContainDocumentWithId(0, SECOND_INDEX_NAME, secondSongId),
+					searchHitContainsFieldWithValue(0, FIELD_TITLE, secondSong.getTitle()),
+					searchHitContainsFieldWithValue(0, FIELD_LYRICS, VALUE_TO_MASKED_VALUE.apply(secondSong.getLyrics())),
+					searchHitContainsFieldWithValue(0, FIELD_ARTIST, secondSong.getArtist()),
+					searchHitContainsFieldWithValue(0, FIELD_STARS, secondSong.getStars())
+			));
+		}
+
+		//DLS
+		try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(TWINS_FIRST_ARTIST_READER)) {
+			String firstSongId = FIND_ID_OF_SONG_WITH_ARTIST.apply(FIRST_INDEX_SONGS_BY_ID, ARTIST_TWINS);
+			String secondSongId = FIND_ID_OF_SONG_WITH_ARTIST.apply(SECOND_INDEX_SONGS_BY_ID, ARTIST_FIRST);
+
+			MultiSearchRequest request = new MultiSearchRequest();
+			request.add(new SearchRequest(FIRST_INDEX_NAME));
+			request.add(new SearchRequest(SECOND_INDEX_NAME));
+
+			MultiSearchResponse response = restHighLevelClient.msearch(request, DEFAULT);
+
+			assertThat(response, isSuccessfulMultiSearchResponse());
+			assertThat(response, numberOfSearchItemResponsesIsEqualTo(2));
+
+			MultiSearchResponse.Item[] responses = response.getResponses();
+
+			assertThat(responses[0].getResponse(), searchHitsContainDocumentWithId(0, FIRST_INDEX_NAME, firstSongId));
+			assertThat(responses[0].getResponse(), searchHitContainsFieldWithValue(0, FIELD_ARTIST, ARTIST_TWINS));
+			assertThat(responses[1].getResponse(), searchHitsContainDocumentWithId(0, SECOND_INDEX_NAME, secondSongId));
+			assertThat(responses[1].getResponse(), searchHitContainsFieldWithValue(0, FIELD_ARTIST, ARTIST_FIRST));
+		}
+	}
```

Comment:
I have a question, should there be tests about multi-get or multi-search via Aliases. I'm not sure if this scenario is supported or not, but if it is then we should add these tests

Reply:
Done: https://github.com/opensearch-project/security/pull/2258/commits/e839ba8793a9db38bf8adf9f2968466455082e6d